### PR TITLE
d1: Update local-development.md

### PR DESCRIPTION
**Pull request by elithrar**
This PR:

- Makes it clearer how to test D1 programmatically (as a start)
- Links out to the `unstable_dev` docs at https://developers.cloudflare.com/workers/wrangler/api/

### CODE REVIEW

**Comment by elithrar** [comment]
Spell check is wrong :)

image

File: content/d1/learning/local-development.md [hunk 1]
```diff
@@ -63,12 +63,91 @@ Refer to the [`wrangler dev` documentation](/workers/wrangler/commands/#dev) to
 
 ## Persist data
 
-**By default, in wrangler `3.0.0` and above, data is persisted across each run of `wrangler dev`**. If your local development and testing requires or assumes an empty database, you should start with a `DROP TABLE <tablename>` statement to delete existing tables before using `CREATE TABLE` to re-create them.
+{{<Aside type="note" heading="Changes in wrangler v3">}}
+
+By default, in wrangler `3.0.0` and above, data is persisted across each run of `wrangler dev`. If your local development and testing requires or assumes an empty database, you should start with a `DROP TABLE <tablename>` statement to delete existing tables before using `CREATE TABLE` to re-create them.
```
Comment (deadlypants1973) [inline content/d1/learning/local-development.md]:
```suggestion
By default, in Wrangler v3 and above, data is persisted across each run of `wrangler dev`. If your local development and testing requires or assumes an empty database, you should start with a `DROP TABLE <tablename>` statement to delete existing tables before using `CREATE TABLE` to re-create them.
```

File: content/d1/learning/local-development.md [hunk 2]
```diff
@@ -63,12 +63,91 @@ Refer to the [`wrangler dev` documentation](/workers/wrangler/commands/#dev) to
...
+
+{{</Aside>}}
 
 Use `wrangler dev --persist-to=/path/to/file` to persist data to a specific location. This can be useful when working in a team (allowing you to share) the same copy, when deploying via CI/CD (to ensure the same starting state), or as a way to keep data when migrating across machines.
 
 Users of wrangler `2.x` must use the `--persist` flag: previous versions of wrangler did not persist data by default.
 
+## Programmatically testing
```
Comment (deadlypants1973) [inline content/d1/learning/local-development.md]:
```suggestion
## Test programmatically
```

File: content/d1/learning/local-development.md [hunk 3]
```diff
@@ -63,12 +63,91 @@ Refer to the [`wrangler dev` documentation](/workers/wrangler/commands/#dev) to
...
+
+Wrangler exposes an [`unstable_dev()`](/workers/wrangler/api/) that allows you to run a local HTTP server for testing Workers and D1. You can run [migrations](/d1/platform/migrations/) against a local database by setting a `preview_database_id` in your `wrangler.toml` configuration.
```
Comment (deadlypants1973) [inline content/d1/learning/local-development.md]:
```suggestion
Wrangler exposes an [`unstable_dev()`](/workers/wrangler/api/) that allows you to run a local HTTP server for testing Workers and D1. Run [migrations](/d1/platform/migrations/) against a local database by setting a `preview_database_id` in your `wrangler.toml` configuration.
```

File: content/d1/learning/local-development.md [hunk 4]
```diff
@@ -63,12 +63,91 @@ Refer to the [`wrangler dev` documentation](/workers/wrangler/commands/#dev) to
...
+```sh
+$ wrangler d1 migrations apply your-database --local
+```
+
+### Usage example
+
+The following example shows how to use Wrangler's `unstable_dev()` API to:
+
+* Run migrations against your local test database, as defined by `preview_database_id`
+* Make a request to an endpoint defined in your Worker - in this example, `/api/users/?limit=2`
```
Comment (deadlypants1973) [inline content/d1/learning/local-development.md]:
```suggestion
* Run migrations against your local test database, as defined by `preview_database_id`.
* Make a request to an endpoint defined in your Worker. This example uses `/api/users/?limit=2`.
```

**Review by deadlypants1973 — APPROVED** [review]

**Pull request merged**